lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -64,7 +66,14 @@ class ThreadRepositoryImpl extends ThreadRepository {
         sort: sort,
         filter: emailFilter?.filter,
         properties: propertiesCreated);
-
+      if (emailFilter?.filterOption != FilterMessageOption.all && networkEmailResponse.emailList!.isNotEmpty) {
```

Comment:
please separate this condition in single method. It should be a comment for easier understand